client/src/modules/AdminCosting/Components/AdminCosting.tsx

```diff
@@ -0,0 +1,138 @@
+import React, { useEffect, useState } from 'react'
+import { Redirect } from 'react-router-dom'
+
+import axios from 'axios'
+
+import { useAuthMandatoryLogin } from '../../../auth/auth_utils'
+
+import '../Styles/Admin.Costing.css'
+
+/** Admin Costing Page
+ * Get costing estimates for GPT4 and GPT3.5 using data from courses that have
+ * at least an x amount of reviews.
+ */
+export const AdminCosting = () => {
+  const [minimumReviews, setMinimumReviews] = useState<number | "">("");
+  const [totalTokens, setTotalTokens] = useState<number>(0)
+  const [totalReviews, setTotalReviews] = useState<number>(0)
+  const [totalChars, setTotalChars] = useState<number>(0)
+  const [totalWords, setTotalWords] = useState<number>(0)
+  const [avgWords, setAvgWords] = useState<number>(0)
+  const [avgChars, setAvgChars] = useState<number>(0)
+  const [avgTokens, setAvgTokens] = useState<number>(0)
+  const [gpt4cost, setGpt4Cost] = useState<number>(0)
+  const [gpt3cost, setGpt3Cost] = useState<number>(0)
+
+  const { isLoggedIn, token, isAuthenticating } = useAuthMandatoryLogin('admin')
+  const [loading, setLoading] = useState(true)
+  const [isAdmin, setIsAdmin] = useState(false)
+
+  useEffect(() => {
+    async function confirmAdmin() {
+      const res = await axios.post(`/api/tokenIsAdmin`, {
+        token: token,
+      })
+
+      if (res.data.result) {
+        setIsAdmin(true)
+      } else {
+        setIsAdmin(false)
+      }
```

Comment:
I think you could condense this into `setIsAdmin(res.data.result)` for easier readability.